qiskit/transpiler/passes/utils/merge_adjacent_barriers.py

```diff
@@ -69,29 +71,21 @@ def run(self, dag):
         if not node_to_barrier_qubits:
             return dag
 
-        # add the merged barriers to a new DAG
-        new_dag = dag.copy_empty_like()
-
-        # go over current nodes, and add them to the new dag
-        for node in dag.topological_op_nodes():
-            if node.name == "barrier":
-                if node in node_to_barrier_qubits:
-                    qubits = node_to_barrier_qubits[node]
-                    # qubits are stored as a set, need to convert to a list
-                    new_dag.apply_operation_back(
-                        Barrier(len(qubits)), qargs=sorted(qubits, key=indices.get), check=False
-                    )
-            else:
-                # copy the condition over too
-                new_dag.apply_operation_back(node.op, node.qargs, node.cargs, check=False)
+        new_dag = copy.copy(dag)
```

Comment:
We don't need to copy the DAG - transpiler passes are allowed to mutate the DAG that's given to them (e.g. you'll see the `BasisTranslator` doing that).  We only really need to construct a new DAG if we _can't_ mutate it in-place for whatever reason - previously there was no method this pass could use that would work, but now `DAGCircuit.replace_block_with_op` lets us do it, so no need to copy.